airflow/models/dag.py

```diff
@@ -897,7 +898,9 @@ def get_num_active_runs(self, external_trigger=None, session=None):
         )
 
         if external_trigger is not None:
-            query = query.filter(DagRun.external_trigger == external_trigger)
+            query = query.filter(
+                DagRun.external_trigger == (expression.true() if external_trigger else expression.false())
```

Comment:
Why is this necessary? What is the type of `external_trigger` here to require this workaround?